pkg/storage/unified/resource/sqlkv.go

```diff
@@ -298,7 +330,194 @@ func (k *sqlKV) TestingSave(ctx context.Context, key string, value []byte) error
 }
 
 func (k *sqlKV) Save(ctx context.Context, section string, key string) (io.WriteCloser, error) {
-	panic("not implemented!")
+	if section == "" {
+		return nil, errors.New("section is required")
+	}
+
+	if section != dataSection && section != eventsSection {
+		return nil, errors.New("invalid section")
+	}
+
+	if key == "" {
+		return nil, errors.New("key is required")
+	}
+
+	return &sqlWriteCloser{
+		kv:      k,
+		ctx:     ctx,
+		section: section,
+		key:     key,
+		buf:     &bytes.Buffer{},
+		closed:  false,
+	}, nil
+}
+
+type sqlWriteCloser struct {
+	kv      *sqlKV
+	ctx     context.Context
+	section string
+	key     string
+	buf     *bytes.Buffer
+	closed  bool
+}
+
+func (w *sqlWriteCloser) Write(value []byte) (int, error) {
+	if w.closed {
+		return 0, errors.New("write to closed writer")
+	}
+
+	return w.buf.Write(value)
+}
+
+func (w *sqlWriteCloser) Close() error {
+	if w.closed {
+		return nil
+	}
+
+	w.closed = true
+
+	// do regular kv save: simple key_path + value insert with conflict check.
```

Comment:
For my own understanding: the conflict check exists to satisfy the KV API, but in the storage backend, we shouldn't have a conflict in the event store, right?